### DISCUSSION TITLE
Set Charge Level

### DISCUSSION THREAD

**Discussion by oderuyter** [alandtse/tesla#270] [General]
Is there a way to set a custom charge level. I have an automation that sets charge level on entering and leaving a zone, however, there is on the option of max range and not maxrange - I would prefer the not maxrange was 80% as opposed to the default of 90% - how can this be achieved?

**Comment by Jonathanese** [comment]
I just now came here for this exact reason. I've seen questions asking something similar in the original integration, and they wanted to focus on bugfixes before new features. I imagine the API changes foiled the original plans.

And to give my own use case, I would want to set the charge level to 50% when I'm at home and 90% while at work. That way, home charging will always keep the battery from getting too low, but then I can still benefit from charging at work when a spot becomes available.

I currently do this manually, and actually had a couple months where I didn't have to charge at home at all. (Free driving!) But in the winter I would still like for it to be plugged it for preconditioning.

One thing you could do in the meantime: I haven't checked how the charging switch works, but you could set up an automation to turn off the charging switch, and have the trigger be when your % charge reaches the desired amount. In my case, I would do the same, but I would use the Tesla location tracker entity.

**Comment by alandtse** [comment]
Please search as this was answered already. https://github.com/alandtse/tesla/discussions/180
